extensions/vscode/src/views/homeView.ts

```diff
@@ -306,14 +306,24 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     }
     try {
       const api = await useApi();
-      await api.files.updateFileList(
+      const apiRequest = api.files.updateFileList(
         activeConfig.configurationName,
         uri,
         action,
         {
           dir: activeDeployment.projectDir,
         },
       );
+      window.withProgress(
+        {
+          title: "Updating File List",
+          location: { viewId: Views.HomeView },
+        },
+        async () => {
+          return apiRequest;
+        },
```

Comment:
Okay its 2 PR's late, but I did try unwrapping these and since it isn't a function it fails to typing 😞

Reply:
:-)